app/models/piece.rb

```diff
@@ -42,8 +42,15 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y) # move and capture
+    @game = game
```

Comment:
One other thing - you should call `valid_move?` so to check if the move is valid or not before doing everything else.  When you have a condition like that where it determines whether or not to do what the method does, you can write it like a "guard clause".  That takes advantage of Ruby's predicate if syntax to exit the method early if a condition is not met.  Something like:

```ruby
return if some_condition
# or
return false if some_condition
```

So, you could do something like:

```ruby
def move_to!(x, y)
  return false if !valid_move?(x, y)
  # rest of the method
end
```

For one final ruby-ism, ruby has an `unless` operator that stands for `if not`, so that could become:\

```ruby
def move_to!(x, y)
  return false unless valid_move?(x, y)
  # rest of the method
end
```

Although `if !valid_move?(...)` works just fine too.

Anyways, you'll probably run into the guard clause pattern again.  Essentially instead of:

```ruby
def my_method
  if some_condition
    # ...
    # ...
    # ...
    # ...
  end
end
```

You have:

```ruby
def my_method
  return unless some_condition
  # ...
  # ...
  # ...
  # ...
end
```